### PR TITLE
[POAE7-2368][Hot Fix] Workaround for Nullability Problem Under NOT IS NULL Expression.

### PR DESCRIPTION
**Pull request by bigPYJ1151**
### What changes were proposed in this pull request?

Workaround for Nullability Problem Under NOT IS NULL Expression.

### Why are the changes needed?

Plan from Velox will always notes nullability as nullable. However, IS NULL expression should be not nullable, which will lead to codegen problem under old data format.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

UTs.

### Which label does this PR belong to?

Bug

Pls take a review @Ma-Jian1

### CODE REVIEW

**Review by Ma-Jian1 — APPROVED** [review]

File: cider/exec/template/LogicalIR.cpp [hunk 1]
```diff
@@ -483,7 +483,13 @@ llvm::Value* CodeGenerator::codegenLogical(const Analyzer::UOper* uoper,
   CHECK(operand_ti.is_boolean());
   const auto operand_lv = codegen(operand, true, co).front();
   CHECK(operand_lv->getType()->isIntegerTy());
-  const bool not_null = (operand_ti.get_notnull() || is_qualified_bin_oper(operand));
+  bool not_null = (operand_ti.get_notnull() || is_qualified_bin_oper(operand));
+  // Workaround for isNull expr.
```
Comment (winningsix) [inline cider/exec/template/LogicalIR.cpp:487]: Do we need a TODO here?

File: cider/exec/template/LogicalIR.cpp [hunk 2]
```diff
@@ -483,7 +483,13 @@ llvm::Value* CodeGenerator::codegenLogical(const Analyzer::UOper* uoper,
...
+  if (auto operand_uoper = dynamic_cast<const Analyzer::UOper*>(operand)) {
+    if (kISNULL == operand_uoper->get_optype()) {
+      not_null = true;
+    }
```
Comment (yma11) [inline cider/exec/template/LogicalIR.cpp:491]: Seems originally `not_null` is determined by `UOper`'s argument, why do we need to fix it to true?

File: cider/exec/template/LogicalIR.cpp (hunk 1 above)
Comment (bigPYJ1151) [reply cider/exec/template/LogicalIR.cpp:487]: TODO what? 🤔

File: cider/exec/template/LogicalIR.cpp (hunk 2 above)
Comment (bigPYJ1151) [reply cider/exec/template/LogicalIR.cpp:491]: The problem was encountered in E2E Test. Due to nullability of expression from Velox always is nullable.

File: cider/exec/template/LogicalIR.cpp (hunk 1 above)
Comment (winningsix) [reply cider/exec/template/LogicalIR.cpp:487]: It's mentioning workaround.

**Review by winningsix — APPROVED** [review]

**Pull request merged**